workshops/geometric_pattern/README.md

```diff
@@ -0,0 +1,538 @@
+
+```js
+function draw() {
+  for (var x = 0; x <= width; x = x+CIRCLE_DIA) {
+    ellipse(x, 0, CIRCLE_DIA, CIRCLE_DIA);
+  }
+}
+```
+
+We're supplying `x` as the x-coordinate (cleverly named, eh?), and 0 as the y-coordinate, of the ellipse's center.
+
+If you save and refresh, you'll see a line of cut-off circles at the top. This is the magic of the for-loop. Since the value of `x` was increased by `CIRCLE_DIA` during every repeated call of the `ellipse()` function, circles were drawn in intervals of `CIRCLE_DIA` pixels.
+
+As to why the circles are cut off -- this is because we set the y-coordinate of the _center_ of every circle to 0.
+
+### Drawing a Grid of Circles
+
+So that's great, we've drawn one row of circles. But what we'd like is to cover the entire canvas with circles.
+
+Just like we used a for-loop to repeat circles in the horizontal direction, we can also use a for-loop to repeat them in the vertical direction.
+
+If we wrap our existing for-loop in another for-loop, we'll be performing the action of filling an entire row with circles, multiple times. And thus covering multiple rows.
+
+Let's put everything so far inside another for-loop:
+
+```js
+function draw() {
+  for (var y = 0; y <= height; y = y+CIRCLE_DIA) {
+    for (var x = 0; x <= width; x = x+CIRCLE_DIA) {
+      ellipse(x, 0, CIRCLE_DIA, CIRCLE_DIA);
+    }
+  }
+}
+```
+
+As with the previous for-loop's `x`, we're executing code as long as the value of `y`, which starts at `0`, is less than or equal to `height`. (`height`, as you may have guessed, is the counterpart to `width`, and it is where p5.js stores the height of the canvas.) We're also incrementing `y` at each iteration by `CIRCLE_DIA`.
+
+Save and refresh Live Preview to check it out!
+
+### Offsetting the Circles in the Y-Direction
+
+We want the circles to overlap, so let's change the spacing in the y-direction.
+
+We can achieve this by changing the incrementing value in the outer for-loop to be `CIRCLE_DIA/2` instead of `CIRCLE_DIA`.
+
+We'll create a variable (call it `CIRCLE_RAD`) for this, at the top of the file, define it in `setup()`, and make the replacement in our outer for-loop, like so:
+
+```js
+var NUM_CIRCLES = 12;
+var CIRCLE_DIA;
+var CIRCLE_RAD;
+
+function setup() {
+  createCanvas(480,600);
+  CIRCLE_DIA = width/NUM_CIRCLES;
+  CIRCLE_RAD = CIRCLE_DIA/2;
+}
+
+function draw() {
+  for (var y = 0; y <= height; y = y+CIRCLE_RAD) {
+    for (var x = 0; x <= width; x = x+CIRCLE_DIA) {
+      ellipse(x, 0, CIRCLE_DIA, CIRCLE_DIA);
+    }
+  }
+}
+```
+
+Save and refresh Live Preview to see the change.
+
+### Offsetting the Circles in the X-Direction
+
+It looks pretty cool right now, but our circles aren't overlapping quite right. It looks like every other row should be shifted over by `CIRCLE_RAD`.
+
+We can keep track of alternating rows by using a flag. A flag is just a variable that stores a `true`/`false` value.
```

Comment:
I didn't want to say the word boolean lol cuz then I'd have to be like "it's named that because some man's name was Alfred Boole" or whatever his name was i don't remember